pyanaconda/modules/payloads/payload/rpm_ostree/utils.py

```diff
@@ -0,0 +1,31 @@
+#
+# Copyright (C) 2020 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from pyanaconda.core import util
+import pyanaconda.errors as errors
+from pyanaconda.payload.errors import PayloadInstallError
+
+__all__ = ["safe_exec_with_redirect"]
+
+
+def safe_exec_with_redirect(cmd, argv, **kwargs):
+    """Like util.execWithRedirect, but treat errors as fatal"""
+    rc = util.execWithRedirect(cmd, argv, **kwargs)
+    if rc != 0:
+        exn = PayloadInstallError("%s %s exited with code %d" % (cmd, argv, rc))
+        if errors.errorHandler.cb(exn) == errors.ERROR_RAISE:
```

Comment:
The error handler cannot be used in the module. You should raise an exception that is mapped to a DBus error. I will open a PR that will use the error handler automatically during the installation, so we don't have to deal with it anymore. `PayloadInstallError` is always fatal, so it should be fine.

I would prefer to eventually remove this function and call `execWithRedirect` directly..

Reply:
Thank you, done.